### ISSUE TITLE
Problems with tracks, google link and kml file content

### ISSUE THREAD

**Issue by bive242** [Dom2364/PlaneAlerter#12]
I'm having some issues with the content in the mails sent.
The map that's included does only show a "bubble"-marker at the first or last position of the aircraft (depending on if the mail is regarding a first or last contact). No tracks are visible if the mail is from when my station lost contact with the aircraft.

The included google-map link is also broken since the coordinates use "," instead of ".".

The attached kml-file is there, but contain no coordinates:
`````
<?xml version="1.0" encoding="UTF-8"?>
<kml xmlns="http://www.opengis.net/kml/2.2">
	<Document>
		<name>OH-LTO</name>
		<Placemark>
			<name>Flight path of OH-LTO</name>
			<Style>
				<LineStyle>
					<color>7fffffff</color>
					<width>2</width>
				</LineStyle>
				<PolyStyle>
					<color>7fffffff</color>
				</PolyStyle>
			</Style>
			<LineString>
				<extrude>1</extrude>
				<altitudeMode>absolute</altitudeMode>
				<coordinates></coordinates>
			</LineString>
		</Placemark>
	</Document>
</kml>
`````
I suspect the problem lies within the handling of "," vs "." in the localization of windows (running latest win10 version).
Default settings before the decimals when using swedish language and localization in windows 10 is "," and not ".".

I'm not familiar with C# (or any kind of development on windows systems) and I've tried to read the code and find the problem. I see that there is some kind of conversion taking place that should replace "," with ".", but it doesn't seem to work with 2.3.0.

I've also tried to change windows language (and any other setting related to language/formats in windows) to english, but the problem persists.

The program is really nice and I hope that this problem is fixable. I found it when I googled around trying to see if someone had done something that could parse and collect data from the json-file that VRS produces.

**Comment by Dom2364** [comment]
I think I found the issue, currently during regular checks trails are not requested to save bandwidth, and only when an alert is sent are the trails requested from VRS. However on last contact alerts the aircraft is already gone from VRS so no trails are returned. Definitely a bit of an oversight when the new trails system was added in 2.3.0. I think the only solution for this is to go back to requesting trails with every check, as any check could be the last set of trails for an aircraft. Unless I or anyone else has a better idea I'll change it back to the old system and release an update soon.

**Comment by bive242** [comment]
Been doing some experimentation myself trying to find the problem. In the general config section of VRS you can set how long it will keep the tracking data, mine was set to 600 seconds. Took a look in the documentation for VRS and from what I can understand it should keep the aircraft in the tracking database during this time to resume tracking if the aircraft appears again within this time. But apparently the json does only contain aircrafts visible in the GUI, not the entire active tracking database.

So request the trails at every check seem to be the only solution if you want the entire track, to save some traffic it might be possible to poll the tracks every other time, keep the track data and if the tracks are empty two consecutive polls assume that the aircraft has disappeared and use the last known tracks?
I'm not bandwidth-limited so for me it doesn't matter if you poll the tracks every time, but I'm sure that there are others how are.

How about the "," vs "." issue? Do you think that can be fixed as well?

**Comment by Dom2364** [comment]
Did a quick test and including full trails increases the aircraftlist.json size by about 9 times, so it definitely needs some sort of bandwidth reduction. I think what I might do is add a setting for how often to request trails, 0 for never/disable trails, 1 for every check, 2 for every other, 3 for every third etc.

I did find the decimal separator in the map URL could be a comma when the map is centred on the user-specified coordinates, so I'll fix that as well. If you can it would help if you reply with the bad map URL to see which parts are formatted wrong. Google Maps seems to accept commas in coodinates so I'm not sure if that's the issue or if it's something else.

**Comment by bive242** [comment]
Like the idea of the setting, probably the best solution to the problem without doing a lot of work.

Here is a bad URL that I just got:
`https://www.google.com.au/maps/search/59,486801,17,37735`

This is how it looks when I click the link. 

![PlaneAlerter-google-error](https://user-images.githubusercontent.com/42555263/165030517-f5750440-25b9-4aa3-9563-38008405e1d0.png)

The message is in Swedish, but it says that it can't find the coordinates. 

If I change the "," to ".", it will work:

![PlaneAlerter-google-working](https://user-images.githubusercontent.com/42555263/165030581-d7105d1c-1478-4edd-8268-75ddbc0503d2.png)


This is the embedded map from the same message. It has some errors according the the warning in the upper right corner of the image.

![staticmap](https://user-images.githubusercontent.com/42555263/165030675-c33875f2-d271-4555-b9fa-3465c7f89cb1.png)

**Comment by Dom2364** [comment]
Ah I see, I think the aircraftlist.json might have the comma separators as they are just using the raw text from the json file.

**Comment by bive242** [comment]
This is a screenshot of a current aircraftlist.json:

![bild](https://user-images.githubusercontent.com/42555263/165033200-00f5464d-2c6e-44a9-9bab-9d734cf36f46.png)

**Comment by Dom2364** [comment]
Ok I found where the commas are coming from, thanks for the screenshots and info, really helps to diagnose these sort of things.

**Comment by bive242** [comment]
Great that you found it :)
And no problem with helping out. I would probably have spent a lot more time developing something similar on linux.
It would be nice to have PlaneAlerter running under mono on a headless linux system, but I guess that requires a lot of changes. Maybe it would be easier if you just could copy over all relevant config files from a windows system to the linux/mono server and thus eliminate the need for a gui?

**Comment by Dom2364** [comment]
I am planning on migrating the program to .NET 5 eventually which should allow it to run on linux without mono but it would still need a GUI to configure it, unless you want to modify the JSON settings files manually.

**Comment by bive242** [comment]
I have no problems with modifying them myself. Been running unix/linux-systems since 1992 (I have the entire linux distribution on 3.5" floppy disks somewhere) and feel more comfortable with unix than with windows :)
I'll be looking forward to trying it out once it's done.

**Comment by Dom2364** [comment]
Resolved in [2.4.0](https://github.com/Dom2364/PlaneAlerter/releases/tag/2.4.0)
